interpreter/host/js.ml

```diff
@@ -198,14 +218,30 @@ let assert_return lits ts at =
   in [], List.flatten (List.rev_map test lits)
 
 let assert_return_nan ts at =
-  let var i = Int32.of_int i @@ at in
-  let init i t = [SetLocal (var i) @@ at] in
-  let test i t =
-    [ GetLocal (var i) @@ at;
-      GetLocal (var i) @@ at;
-      Compare (eq_of t) @@ at;
+  let test t =
```

Comment:
The previous version was intended to work correctly for multiple return values. I believe that dropping the auxiliary locals will consume them in the wrong order, inconsistent with `ts`. (Though instead of using locals you can probably reverse `ts`.)